addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -260,6 +260,10 @@ export class PaymentScreen extends Component {
         if (!this.check_cash_rounding_has_been_well_applied()) {
             return;
         }
+        const linesToRemove = this.currentOrder.lines.filter((line) => line.qty === 0);
```

Comment:
```suggestion
        const linesToRemove = this.currentOrder.lines.filter((line) => floatIsZero(line.qty, decimals));
```

For the `decimals`, find it from the uom of the product.